### PR TITLE
Add elasticsearch service

### LINKED ISSUE #36: Add elasticsearch service
Reference: https://github.com/cachix/devenv/blob/main/src/modules/services/elasticsearch.nix

### PR DESCRIPTION
**Pull request by shivaraj-bh**
Resolves #36 
- [x] Add test

### CODE REVIEW

File: nix/elasticsearch.nix [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+{
+  options = {
+    enable = lib.mkEnableOption name;
+
+    package = lib.mkPackageOption pkgs "elasticsearch7" { };
+
+    dataDir = lib.mkOption {
+      type = types.str;
+      # TODO: elasticsearch requires absolute path, is there a way to avoid this?
+      default = "$PWD/data/${name}";
```
Comment (shivaraj-bh) [inline nix/elasticsearch.nix:24]: I am not sure if this is the right way to go about this, but it works.

Comment (srid) [reply nix/elasticsearch.nix:24]: Why not compute absolute path in the script?

```bash
 export ES_HOME="realpath ${config.dataDir}"
```

File: nix/elasticsearch.nix [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# Based on https://github.com/cachix/devenv/blob/main/src/modules/services/elasticsearch.nix
+# Note: 
+#   To use elastic search you will need the following inside your `perSystem`, if not already configured:
+#   { 
+#     _module.args.pkgs = import inputs.nixpkgs {
+#        inherit system;
+#        # Required for elastic search
+#        config.allowUnfree = true;
+#      };
+#   }
```
Comment (srid) [inline nix/elasticsearch.nix:10]: This could be put in submodule's `description`?

(nix-browser can render it ultimately, so can flake.parts website)

**Comment by shivaraj-bh** [comment]
Will be adding native test to this PR once https://github.com/juspay/services-flake/pull/38 is merged

**Review by srid — APPROVED** [review]

**Pull request merged**